### PR TITLE
New Scala cheatsheet

### PR DESCRIPTION
**Pull request by amitdev**
**What does your Instant Answer do?**
It is a cheat sheet for Scala programming language.

**What is the data source for your Instant Answer? (Provide a link if possible)**
http://docs.scala-lang.org/cheatsheets/ with some minor corrections to make it simpler.

**Why did you choose this data source?**
This is part of the official Scala documentation.

**Are there any other alternative (better) data sources?**
[There are](https://duckduckgo.com/?q=scala+cheat+sheet) some other data sources, but the above one seems to be better. We could add missing topics from other data sources also to improve this.

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Programmers.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
Not that I'm aware of - saw this in the [wiki](https://github.com/duckduckgo/duckduckgo/wiki/Programming-IA-Coverage).

**Are you having any problems? Do you need our help with anything?**
This is super simple stuff - looks pretty clear.

**Where did you hear about DuckDuckHack? (For first time contributors)**
I've been a user of ddg search.

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
scala

-----
IA Page: https://duck.co/ia/view/scala_cheat_sheet

### CODE REVIEW

**Comment by moollaza** [comment]
@amitdev thanks for your contribution!

Can you please fill out the [pull request template](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/pull_request_template_cheat_sheet.md) and edit your initial comment. Once completed, we'll be sure to review your PR and give you some feedback.

Thanks!

File: share/goodie/cheat_sheets/json/scala.json [hunk 1]
```diff
@@ -0,0 +1,131 @@
+{
+    "id": "scala_cheat_sheet",
+    "name": "Scala",
+    "metadata": {
+        "sourceName": "Scala docs",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/scala.json]: Please change this to `Scala-Lang` to reflect the domain name

**Comment by amitdev** [comment]
@moollaza, added the PR template. Since I did not see a PR template for cheat sheet in the docs, added a link there too - duckduckgo/duckduckgo-documentation#339.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Scala Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/scala_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Scala Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/scala_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
@amitdev Thanks! This LGTM :+1:

**Comment by moollaza** [comment]
Apologies for the delay!

**Comment by moollaza** [comment]
@amitdev congrats! It's live: https://duckduckgo.com/?q=Scala%20Cheat%20Sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Pull request merged**